pkg/querier/queryrange/results_cache.go

```diff
@@ -416,7 +417,8 @@ func toExtent(ctx context.Context, req Request, res Response) (Extent, error) {
 }
 
 // partition calculates the required requests to satisfy req given the cached data.
-func partition(req Request, extents []Extent, extractor Extractor) ([]Request, []Response, error) {
+// extents must be in order by start time.
+func partition(req Request, extents []Extent, extractor Extractor, minCacheExtent int64) ([]Request, []Response, error) {
```

Comment:
Can you make this a function on `(s resultsCache)` so we don't have to pass `extractor`?

And if `minCacheExtent`, can we stick it in this function / in the package so we don't have so many args?

edit: I see its for testing.  Can we make it a field?   I just think so many args is a bit of smell.